src/Rtc/Create.tsx

```diff
@@ -167,18 +167,41 @@ const Create = ({
           // isVideoEnabledRef checks if the permission is already taken once
           if (!isVideoEnabledRef.current) {
             try {
-              // This creates local audio and video track
-              await engine.current?.enableVideo();
-              isVideoEnabledRef.current = true;
-            } catch (error) {
+              await engine.current.enableVideo();
               dispatch({
                 type: 'LocalMuteAudio',
-                value: [ToggleState.disabled],
+                value: [ToggleState.enabled],
               });
               dispatch({
                 type: 'LocalMuteVideo',
-                value: [ToggleState.disabled],
+                value: [ToggleState.enabled],
               });
+            } catch (e) {
+              if ((e as any).status) {
```

Comment:
destructure the error object for better readability. 

 `const {status} = error`
 
 `const {audioError, videoError} = status`
 
 Then have the respective checks for audio and video undefined and their respective logic.

Reply:
Solved in the latest commit